### ISSUE TITLE
Not compatible with 'com.google.android.gms:play-services-base:17.1.0'

### ISSUE THREAD

**Issue by Mohsin92** [OneSignal/OneSignal-Android-SDK#929]
**Description:**

(your description here)

**Environment**

**Steps to Reproduce Issue:**

1. (your steps here)
2.
3.

**Anything else:**

(crash stacktraces, as well as any other information here)

**Comment by Mohsin92** [comment]
compile with 'com.google.android.gms:play-services-base:17.1.0'
its throws error for conflicts with existing firebase libraries.
please look into this issue.

**Comment by jkasten2** [comment]
@Mohsin92 To fix the issue make sure to follow these to steps;
1. Enable Jetifier
    - https://developer.android.com/jetpack/androidx/migrate
2. Ensure you have correctly added the OneSignal-Gradle-Plugin to your project
   - https://documentation.onesignal.com/docs/android-sdk-setup#section-1-adding-one-signal-dependencies

**More details**
OneSignal depends on `com.google.android.gms:play-services-base:[10.2.1, 16.1.99]` which  version `17.1.0` is outside of the version range of our defined depepency. We defined this limit as going over introduces a dependent switch from Android Support Library to AndroidX by the `play-services-base` library. OneSignal has not switched to AndroidX yet but Google's Jetifier does resolve this issue as an bridge as all libraries would not be able to move over at the same time.

 A lot of Google libraries depend on `play-services-base`, upgrading this without upgrading to a compatible version of each parent you use will result in compile or runtime issues. [MVN repository indicates 303 libraries](https://mvnrepository.com/artifact/com.google.android.gms/play-services-base?repo=google) that depend on this.
We created a Gradle plugin to automatically fix issues with mixing Google libraries as Gradle does not have built in handling to auto upgrade parent libraries.  [OneSignal-Gradle-Plugin](https://github.com/OneSignal/OneSignal-Gradle-Plugin)
